plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -2686,3 +2686,119 @@ function wc_cache_get_multiple( $keys, $group = '', $force = false ) {
 	}
 	return $values;
 }
+
+/**
+ * Delete multiple transients in a single operation.
+ *
+ * IMPORTANT: This is a private function (internal use ONLY).
+ *
+ * This function efficiently deletes multiple transients at once, using a direct
+ * database query when possible for better performance.
+ *
+ * @internal
+ *
+ * @since 9.8.0
+ * @param array $transients Array of transient names to delete (without the '_transient_' prefix).
+ * @return bool True on success, false on failure.
+ */
+function _wc_delete_transients( $transients ) {
+	global $wpdb;
+
+	if ( empty( $transients ) || ! is_array( $transients ) ) {
+		return false;
+	}
+
+	// If using external object cache, delete each transient individually.
+	if ( wp_using_ext_object_cache() ) {
+		foreach ( $transients as $transient ) {
+			delete_transient( $transient );
+		}
+		return true;
+	} else {
+		// For database storage, create a list of transient option names.
+		$transient_names = array();
+		foreach ( $transients as $transient ) {
+			$transient_names[] = '_transient_' . $transient;
+			$transient_names[] = '_transient_timeout_' . $transient;
+		}
+
+		// Limit the number of items in a single query to avoid exceeding database query parameter limits.
+		if ( count( $transient_names ) > 999 ) {
+			// Process in smaller chunks of 200 option names (100 transients) to reduce memory usage.
+			$chunks  = array_chunk( $transients, 100 );
```

Comment:
It's weird allowing 999 at once but for bigger values we chunk them per 100. Should we check for `> 100`?

Reply:
I think you're right in pointing out that we should reduce 999 as its quite a lot of records to process in one operation, and previously we were batching by 500 but that wasn't very efficient in terms of memory after testing.

What do you think about reducing 999 to 499?

Reply:
Also to address your `> 100` question, I guess I was trying to balance memory vs efficiency. I'm open to changing the batch amount, or to threshold but I felt like batching 100 at a time was pretty safe above 499 transient names (or 250 transients).

Reply:
https://github.com/woocommerce/woocommerce/pull/55931/commits/9d9820c77eb5ebc81048c4f48880ec53157da08b

Reply:
I followed up in https://github.com/woocommerce/woocommerce/pull/55931#discussion_r1975186095 🤦